src/components/MultiGestureCanvas/usePinchGesture.ts

```diff
@@ -0,0 +1,186 @@
+    // we need to have extra "bounce" translation variables
+    const pinchBounceTranslateX = useSharedValue(0);
+    const pinchBounceTranslateY = useSharedValue(0);
+
+    // Update the total (pinch) translation based on the regular pinch + bounce
+    useAnimatedReaction(
+        () => [pinchTranslateX.value, pinchTranslateY.value, pinchBounceTranslateX.value, pinchBounceTranslateY.value],
+        ([translateX, translateY, bounceX, bounceY]) => {
+            totalPinchTranslateX.value = translateX + bounceX;
+            totalPinchTranslateY.value = translateY + bounceY;
+        },
+    );
+
+    /**
+     * Calculates the adjusted focal point of the pinch gesture,
+     * based on the canvas size and the current offset
+     */
+    const getAdjustedFocal = MultiGestureCanvasUtils.useWorkletCallback(
+        (focalX: number, focalY: number) => ({
+            x: focalX - (canvasSize.width / 2 + offsetX.value),
+            y: focalY - (canvasSize.height / 2 + offsetY.value),
+        }),
+        [canvasSize.width, canvasSize.height],
+    );
+
+    // The pinch gesture is disabled when we release one of the fingers
+    // On the next render, we need to re-enable the pinch gesture
+    const [pinchEnabled, setPinchEnabled] = useState(true);
+    useEffect(() => {
+        if (pinchEnabled) {
+            return;
+        }
+        setPinchEnabled(true);
+    }, [pinchEnabled]);
+
+    const pinchGesture = Gesture.Pinch()
+        .enabled(pinchEnabled)
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        .onTouchesDown((_evt, state) => {
+            // We don't want to activate pinch gesture when we are swiping in the pager
+            if (!isSwipingInPager.value) {
+                return;
+            }
+
+            state.fail();
+        })
+        .onStart((evt) => {
+            stopAnimation();
+
+            // Set the origin focal point of the pinch gesture at the start of the gesture
+            const adjustedFocal = getAdjustedFocal(evt.focalX, evt.focalY);
+            pinchOrigin.x.value = adjustedFocal.x;
+            pinchOrigin.y.value = adjustedFocal.y;
+        })
+        .onChange((evt) => {
+            // Disable the pinch gesture if one finger is released,
+            // to prevent the content from shaking/jumping
+            if (evt.numberOfPointers !== 2) {
+                runOnJS(setPinchEnabled)(false);
+                return;
+            }
+
+            const newZoomScale = pinchScale.value * evt.scale;
+
+            // Limit the zoom scale to zoom range including bounce range
+            if (
+                zoomScale.value >= zoomRange.min * MultiGestureCanvasUtils.zoomScaleBounceFactors.min &&
+                zoomScale.value <= zoomRange.max * MultiGestureCanvasUtils.zoomScaleBounceFactors.max
+            ) {
+                zoomScale.value = newZoomScale;
+                currentPinchScale.value = evt.scale;
+
+                if (onScaleChanged != null) {
```

Comment:
Should it be like this?
```suggestion
                if (onScaleChanged !== null) {
```